snowflake/cmd/dropDetection.go

```diff
@@ -0,0 +1,294 @@
+	}
+
+	queryBuilder.WriteString(`GROUP BY
+	sourceIP,
+	sourcePodName,
+	sourcePodNamespace,
+	destinationIP,
+	destinationPodName,
+	destinationPodNamespace,
+	flowStartDate,
+	ingressNetworkPolicyRuleAction,
+	egressNetworkPolicyRuleAction
+), processed_flows AS (
+SELECT
+CASE WHEN f.ingressNetworkPolicyRuleAction = 2 OR f.ingressNetworkPolicyRuleAction = 3
+	THEN
+	(CASE WHEN f.destinationPodName IS NOT NULL
+		THEN CONCAT(f.destinationPodNamespace, '/', f.destinationPodName)
+		ELSE f.destinationIP
+	END)
+ELSE
+	(CASE WHEN f.sourcePodName IS NOT NULL
+		THEN CONCAT(f.sourcePodNamespace, '/', f.sourcePodName)
+		ELSE f.sourceIP
+	END)
+END as endpoint,
+CASE WHEN f.ingressNetworkPolicyRuleAction = 2 OR f.ingressNetworkPolicyRuleAction = 3
+	THEN
+	'ingress'
+ELSE
+	'egress'
+END as direction,
+f.flowStartDate as date,
+f.flowNumber as dropNumber
+FROM filtered_flows AS f
+`)
+
+	// Aggregate traffic drop number for each endpoint and direction
+	queryBuilder.WriteString(`), aggregated_flows as (
+SELECT 
+	pf.endpoint,
+	pf.direction,
+	pf.date, 
+	SUM(pf.dropNumber) as dropNumber
+FROM processed_flows AS pf
+GROUP BY
+	pf.endpoint,
+	pf.direction,
+	pf.date
+`)
+
+	// Choose the endpoint + direction as the partition field for the dropDetection UDTF
+	// because traffic drop detection is run on ingress and egress direction for each endpoint.
+	functionName := udfs.GetFunctionName(dropDetectionFunctionName, functionVersion)
+	fmt.Fprintf(&queryBuilder, `)
+SELECT 
+	r.job_type, 
+	r.detection_id, 
+	r.time_created, 
+	r.endpoint,
+	r.direction,
+	r.avg_drop,
+	r.stdev_drop,
+	r.anomaly_drop_date,
+	r.anomaly_drop_number FROM aggregated_flows AS af,
+TABLE(%s(
+	'%s',
+	'%s',
+	af.endpoint,
+	af.direction,
+	af.date, 
+	af.dropNumber
+) over (partition by af.endpoint, af.direction)) as r
+`, functionName, jobType, detectionID)
+
+	return queryBuilder.String(), nil
+}
+
+// dropDetectionCmd represents the drop-detection command
+var dropDetectionCmd = &cobra.Command{
+	Use:   "drop-detection",
+	Short: "Run the abnormal traffic drop detection UDF in Snowflake",
+	Long: `This command runs the abnormal traffic drop detection UDF in
+Snowflake. You need to bring your own Snowflake account and run the onboard
+command first.
+
+Run abnormal traffic drop detection with default configuration on database ANTREA_C9JR8KUKUIV4R72S:
+"theia-sf drop-detection --database-name ANTREA_C9JR8KUKUIV4R72S"
+
+The "drop-detection" command requires a Snowflake warehouse to run abnormal
+traffic drop detection UDF in Snowflake. By default, it will create a temporary
+one.
+You can also bring your own by using the "--warehouse-name" parameter.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		jobType, _ := cmd.Flags().GetString("type")
+		if jobType != "initial" && jobType != "periodical" {
+			return fmt.Errorf("invalid --type argument")
+		}
```

Comment:
I think it would be better to reject anything that's not `"initial"`, since this is the only thing we support at the moment

Reply:
Sounds good, I made this change for policy-recommendation too.